Build failure: ouch

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *ouch* on x86_64 darwin

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/gb71wm0zqwxhaa8ipyqxkmalxwl169c6-source
source root is source
Executing cargoSetupPostUnpackHook
unpacking source archive /nix/store/14csnwc17dk78l0dmlwkfw50klqrwmpy-ouch-0.5.0-vendor.tar.gz
Finished cargoSetupPostUnpackHook
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
Executing cargoSetupPostPatchHook
Validating consistency between /private/tmp/nix-build-ouch-0.5.0.drv-0/source/Cargo.lock and /private/tmp/nix-build-ouch-0.5.0.drv-0/ouch-0.5.0-vendor.tar.gz/Cargo.lock
Finished cargoSetupPostPatchHook
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Executing cargoBuildHook
++ env CC_X86_64_APPLE_DARWIN=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc CXX_X86_64_APPLE_DARWIN=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/c++ CARGO_TARGET_X86_64_APPLE_DARWIN_LINKER=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc CC_X86_64_APPLE_DARWIN=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc CXX_X86_64_APPLE_DARWIN=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/c++ CARGO_TARGET_X86_64_APPLE_DARWIN_LINKER=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc CARGO_BUILD_TARGET=x86_64-apple-darwin HOST_CC=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc HOST_CXX=/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/c++ cargo build -j 12 --target x86_64-apple-darwin --frozen --profile release --features=zstd/pkg-config
   Compiling libc v0.2.150
   Compiling cfg-if v1.0.0
   Compiling pkg-config v0.3.27
   Compiling autocfg v1.1.0
   Compiling proc-macro2 v1.0.69
   Compiling unicode-ident v1.0.12
   Compiling scopeguard v1.2.0
   Compiling utf8parse v0.2.1
   Compiling heck v0.4.1
   Compiling anstyle-query v1.0.0
   Compiling anstyle v1.0.4
   Compiling version_check v0.9.4
   Compiling anstyle-parse v0.2.2
   Compiling memchr v2.6.4
   Compiling crossbeam-utils v0.8.16
   Compiling typenum v1.17.0
   Compiling colorchoice v1.0.0
   Compiling strsim v0.10.0
   Compiling vcpkg v0.2.15
   Compiling anstream v0.6.4
   Compiling memoffset v0.9.0
   Compiling generic-array v0.14.7
   Compiling crossbeam-epoch v0.9.15
   Compiling clap_lex v0.6.0
   Compiling lock_api v0.4.11
   Compiling aho-corasick v1.1.2
   Compiling clap_builder v4.4.9
   Compiling time-core v0.1.2
   Compiling quote v1.0.33
   Compiling crc32fast v1.3.2
   Compiling syn v2.0.39
   Compiling powerfmt v0.2.0
   Compiling regex-syntax v0.8.2
   Compiling time-macros v0.2.15
   Compiling jobserver v0.1.27
   Compiling deranged v0.3.9
   Compiling cc v1.0.83
   Compiling adler v1.0.2
   Compiling byteorder v1.5.0
   Compiling futures-core v0.3.29
   Compiling crossbeam-deque v0.8.3
   Compiling miniz_oxide v0.7.1
   Compiling filetime v0.2.22
   Compiling time v0.3.30
   Compiling getrandom v0.2.11
   Compiling rayon-core v1.12.0
   Compiling thiserror v1.0.50
   Compiling roff v0.2.1
   Compiling rustix v0.38.25
   Compiling zstd-safe v7.0.0
   Compiling snap v1.1.0
   Compiling libz-sys v1.1.12
   Compiling zstd-sys v2.0.9+zstd.1.5.5
   Compiling unrar_sys v0.3.1
   Compiling lzma-sys v0.1.20
   Compiling bzip2-sys v0.1.11+1.0.8
   Compiling regex-automata v0.4.3
   Compiling flate2 v1.0.28
   Compiling digest v0.9.0
   Compiling block-buffer v0.9.0
   Compiling nanorand v0.7.0
   Compiling spin v0.9.8
   Compiling num_cpus v1.16.0
   Compiling errno v0.3.7
   Compiling bstr v1.8.0
   Compiling fs-err v2.11.0
   Compiling bit-vec v0.6.3
   Compiling crc-catalog v2.4.0
   Compiling opaque-debug v0.3.0
   Compiling futures-sink v0.3.29
   Compiling same-file v1.0.6
   Compiling bitflags v2.4.1
   Compiling static_assertions v1.1.0
   Compiling cpufeatures v0.2.11
   Compiling log v0.4.20
   Compiling bit-set v0.5.3
   Compiling sha2 v0.9.9
   Compiling walkdir v2.4.0
   Compiling twox-hash v1.6.3
   Compiling crc v3.0.1
   Compiling core_affinity v0.8.1
   Compiling nt-time v0.5.0
   Compiling filetime_creation v0.1.6
   Compiling lzma-rust v0.1.5
   Compiling regex v1.10.2
   Compiling globset v0.4.14
   Compiling xattr v1.0.1
   Compiling bytes v1.5.0
   Compiling either v1.9.0
   Compiling clap_derive v4.4.7
   Compiling pin-project-internal v1.1.3
   Compiling thiserror-impl v1.0.50
   Compiling bitflags v1.3.2
   Compiling widestring v1.0.2
   Compiling fastrand v2.0.1
   Compiling rayon v1.8.0
   Compiling tempfile v3.8.1
   Compiling ignore v0.4.21
   Compiling tar v0.4.40
   Compiling sevenz-rust v0.5.3
   Compiling pin-project v1.1.3
   Compiling flume v0.10.14
   Compiling bzip2 v0.4.4
   Compiling xz2 v0.1.7
   Compiling zstd v0.13.0
   Compiling gzp v0.11.3
   Compiling lz4_flex v0.11.1
   Compiling clap v4.4.10
   Compiling zip v0.6.6
   Compiling clap_complete v4.4.4
   Compiling clap_mangen v0.2.15
   Compiling atty v0.2.14
   Compiling once_cell v1.18.0
   Compiling linked-hash-map v0.5.6
   Compiling ouch v0.5.0 (/private/tmp/nix-build-ouch-0.5.0.drv-0/source)
   Compiling unrar v0.5.2
[0m38;5;9merror: linking with `/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc` failed: exit status: 1
  [0m38;5;12m|
  [0m38;5;12m= note: LC_ALL="C" PATH="/nix/store/w75hzbshzj3x5a7wgjcw3s8ax0xi86zn-rustc-1.73.0/lib/rustlib/x86_64-apple-darwin/bin:/nix/store/y0hc7ajswm1yh0s740irkx3pbva79l8d-cargo-1.73.0/bin:/nix/store/y1ix1d3mzf75xbv13y2van8b2wrvgqza-cargo-auditable-0.6.1/bin:/nix/store/4jm328xy6xbmph9l4f6xlzigdac0i8ch-pkg-config-wrapper-0.29.2/bin:/nix/store/90qhxqgb9ygkkh4mv02blk6n01cxv2n9-auditable-cargo-1.73.0/bin:/nix/store/y0hc7ajswm1yh0s740irkx3pbva79l8d-cargo-1.73.0/bin:/nix/store/w75hzbshzj3x5a7wgjcw3s8ax0xi86zn-rustc-1.73.0/bin:/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin:/nix/store/jfl9vb9rbxvsfrk79bf03xdj6d77x9ba-clang-16.0.6/bin:/nix/store/vd0b59wk8x56i8kmwda8jl5h37wnixn2-coreutils-9.3/bin:/nix/store/dm67kzmf9jscndd56cvxvlx48ksicwsn-cctools-binutils-darwin-wrapper-16.0.6-973.0.1/bin:/nix/store/nxzsfcz92w87i838a8p35b0iyc19m796-cctools-binutils-darwin-16.0.6-973.0.1/bin:/nix/store/vd0b59wk8x56i8kmwda8jl5h37wnixn2-coreutils-9.3/bin:/nix/store/cmh1dkqbzds5ilgmz9vc7jpavd8lflp3-findutils-4.9.0/bin:/nix/store/2jp91fvgj8mjnpx2vjavc7sw5njbdv04-diffutils-3.10/bin:/nix/store/lpzfs8kvdf7fs2j2fbc8a0r2zjjlygna-gnused-4.9/bin:/nix/store/ccnn45f8fjs7dcj53jgw8jmaa2dq9nrm-gnugrep-3.11/bin:/nix/store/f27mhm2kpk37mpj88f77j0y2ncgb95a7-gawk-5.2.2/bin:/nix/store/9zmlp4a3y519q3hkflyik9alsmjzpyda-gnutar-1.35/bin:/nix/store/q0vax9kk0mylpd91d9j18556awm1l6ip-gzip-1.13/bin:/nix/store/iy9w0wfxccw0b4lmgkd4fgf2kyq1px6j-bzip2-1.0.8-bin/bin:/nix/store/x5dy4sqg1nxsl7nwibanfrbx95wdb228-gnumake-4.4.1/bin:/nix/store/0ly5lpwanagc5qpmnqrw8aqxb4ivw7kf-bash-5.2-p15/bin:/nix/store/a4737w7v4pbjg539bc2jd7ixsdp66fg3-patch-2.7.6/bin:/nix/store/pd3w9r6qz79kivw2sljcvkwnnsxpi7cd-xz-5.4.4-bin/bin:/nix/store/dl50w02kxffqn52rc1hlps01danyac5g-file-5.45/bin" VSLANG="1033" ZERO_AR_DATE="1" "/nix/store/dgmdis15d7s7vlkfkz5w5b2wnna2syfz-clang-wrapper-16.0.6/bin/cc" "-arch" "x86_64" "-m64" "/private/tmp/nix-build-ouch-0.5.0.drv-0/rustcyR6ze3/symbols.o" "/private/tmp/nix-build-ouch-0.5.0.drv-0/source/target/x86_64-apple-darwin/release/deps/ouch-4020992a35d83fbf.ouch.60be277a122b23df-cgu.0.rcgu.o" "-L""/private/tmp/nix-build-ouch-0.5.0.drv-0/source/target/x86_64-apple-darwin/release/deps" "-L" "/private/tmp/nix-build-ouch-0.5.0.drv-0/source/target/release/deps" "-L" "/nix/store/5imwckilwsg7svylr527zyignv3kjvk4-bzip2-1.0.8/lib" "-L" "/nix/store/ig962y9d643hwwx9xqhdadc1lll8gvig-zlib-1.3/lib" "-L" "/private/tmp/nix-build-ouch-0.5.0.drv-0/source/target/x86_64-apple-darwin/release/build/unrar_sys-66022c73b79e2662/out""-L" "/nix/store/r195pxi80cy1nn56vfyr4zcbblbyk9n1-xz-5.4.4/lib" "-L" "/nix/store/6kv3r5manl5v60cnj0rj73f1nr4a9vg7-zstd-1.5.5/lib" "-L" "/nix/store/w75hzbshzj3x5a7wgjcw3s8ax0xi86zn-rustc-1.73.0/lib/rustlib/x86_64-apple-darwin/lib" "/private/tmp/nix-build-ouch-0.5.0.drv-0/rustcyR6ze3/libunrar_sys-a73d56025a07d327.rlib" "/nix/store/w75hzbshzj3x5a7wgjcw3s8ax0xi86zn-rustc-1.73.0/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-69674078d75f0362.rlib" "-lzstd" "-llzma" "-lbz2" "-lz" "-lc++" "-lpthread" "-liconv" "-lSystem" "-lc" "-lm" "-L" "/nix/store/w75hzbshzj3x5a7wgjcw3s8ax0xi86zn-rustc-1.73.0/lib/rustlib/x86_64-apple-darwin/lib" "-o" "/private/tmp/nix-build-ouch-0.5.0.drv-0/source/target/x86_64-apple-darwin/release/deps/ouch-4020992a35d83fbf" "-Wl,-dead_strip" "-nodefaultlibs" "/private/tmp/nix-build-ouch-0.5.0.drv-0/source/target/x86_64-apple-darwin/release/deps/ouch_audit_data.o" "-Wl,-u,_AUDITABLE_VERSION_INFO"
  [0m38;5;12m= note: Undefined symbols for architecture x86_64:
            "std::bad_alloc::~bad_alloc()", referenced from:
                ModelPPM::RestartModelRare() in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
                FragmentedWindow::Init(unsigned long) in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
                Unpack::Init(unsigned long, bool) in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
            "vtable for __cxxabiv1::__enum_type_info", referenced from:
                typeinfo for RAR_EXIT in libunrar_sys-a73d56025a07d327.rlib(errhnd.o)
            NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
            "operator new[](unsigned long)", referenced from:
                Unpack::InitMT() in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
                QuickOpen::Init(Archive*, bool) in libunrar_sys-a73d56025a07d327.rlib(qopen.o)
                QuickOpen::QuickOpen() in libunrar_sys-a73d56025a07d327.rlib(qopen.o)
                RarVM::Init() in libunrar_sys-a73d56025a07d327.rlib(rarvm.o)
                BitInput::BitInput(bool) in libunrar_sys-a73d56025a07d327.rlib(getbits.o)
            "vtable for __cxxabiv1::__si_class_type_info", referenced from:
                typeinfo for RarCheckPassword in libunrar_sys-a73d56025a07d327.rlib(extract.o)
                typeinfo for Archive in libunrar_sys-a73d56025a07d327.rlib(archive.o)
            NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
            "operator delete[](void*)", referenced from:
                Unpack::InitMT() in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
                Unpack::~Unpack() in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
                QuickOpen::Init(Archive*, bool) in libunrar_sys-a73d56025a07d327.rlib(qopen.o)
                QuickOpen::~QuickOpen() in libunrar_sys-a73d56025a07d327.rlib(qopen.o)
                RarVM::~RarVM() in libunrar_sys-a73d56025a07d327.rlib(rarvm.o)
                BitInput::~BitInput() in libunrar_sys-a73d56025a07d327.rlib(getbits.o)
                BitInput::SetExternalBuffer(unsigned char*) in libunrar_sys-a73d56025a07d327.rlib(getbits.o)
                ...
            "vtable for __cxxabiv1::__class_type_info", referenced from:
                typeinfo for CheckPassword in libunrar_sys-a73d56025a07d327.rlib(extract.o)
                typeinfo for File in libunrar_sys-a73d56025a07d327.rlib(file.o)
            NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
            "std::bad_alloc::bad_alloc()", referenced from:
                ModelPPM::RestartModelRare() in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
                FragmentedWindow::Init(unsigned long) in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
                Unpack::Init(unsigned long, bool) in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
            "___cxa_allocate_exception", referenced from:
                std::__1::__throw_length_error[abi:v160006](char const*) in libunrar_sys-a73d56025a07d327.rlib(extract.o)
                ErrorHandler::MemoryError() in libunrar_sys-a73d56025a07d327.rlib(errhnd.o)
                ErrorHandler::Exit(RAR_EXIT) in libunrar_sys-a73d56025a07d327.rlib(errhnd.o)
                ErrorHandler::ReadError(wchar_t const*) in libunrar_sys-a73d56025a07d327.rlib(errhnd.o)
                ErrorHandler::WriteError(wchar_t const*, wchar_t const*) in libunrar_sys-a73d56025a07d327.rlib(errhnd.o)
                ErrorHandler::SeekError(wchar_t const*) in libunrar_sys-a73d56025a07d327.rlib(errhnd.o)
                ModelPPM::RestartModelRare() in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
                ...
            "std::length_error::~length_error()", referenced from:
                std::__1::__throw_length_error[abi:v160006](char const*) in libunrar_sys-a73d56025a07d327.rlib(extract.o)
            "std::terminate()", referenced from:
                ___clang_call_terminate in libunrar_sys-a73d56025a07d327.rlib(extract.o)
            "___cxa_free_exception", referenced from:
                std::__1::__throw_length_error[abi:v160006](char const*) in libunrar_sys-a73d56025a07d327.rlib(extract.o)
            "typeinfo for std::bad_alloc", referenced from:
                GCC_except_table1 in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                GCC_except_table8 in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                ModelPPM::RestartModelRare() in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
                FragmentedWindow::Init(unsigned long) in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
                Unpack::Init(unsigned long, bool) in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
            "___gxx_personality_v0", referenced from:
                _RAROpenArchiveEx in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                DataSet::DataSet() in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                _RARCloseArchive in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                _RARReadHeaderEx in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                ProcessFile(void*, int, char*, char*, wchar_t*, wchar_t*) in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                CmdExtract::CmdExtract(CommandData*) in libunrar_sys-a73d56025a07d327.rlib(extract.o)
                CmdExtract::~CmdExtract() in libunrar_sys-a73d56025a07d327.rlib(extract.o)
                ...
            "___cxa_throw", referenced from:
                std::__1::__throw_length_error[abi:v160006](char const*) in libunrar_sys-a73d56025a07d327.rlib(extract.o)
                ErrorHandler::MemoryError() in libunrar_sys-a73d56025a07d327.rlib(errhnd.o)
                ErrorHandler::Exit(RAR_EXIT) in libunrar_sys-a73d56025a07d327.rlib(errhnd.o)
                ErrorHandler::ReadError(wchar_t const*) in libunrar_sys-a73d56025a07d327.rlib(errhnd.o)
                ErrorHandler::WriteError(wchar_t const*, wchar_t const*) in libunrar_sys-a73d56025a07d327.rlib(errhnd.o)
                ErrorHandler::SeekError(wchar_t const*) in libunrar_sys-a73d56025a07d327.rlib(errhnd.o)
                ModelPPM::RestartModelRare() in libunrar_sys-a73d56025a07d327.rlib(unpack.o)
                ...
            "typeinfo for std::length_error", referenced from:
                std::__1::__throw_length_error[abi:v160006](char const*) in libunrar_sys-a73d56025a07d327.rlib(extract.o)
            "vtable for std::length_error", referenced from:
                std::length_error::length_error[abi:v160006](char const*) in libunrar_sys-a73d56025a07d327.rlib(extract.o)
            NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
            "operator delete(void*)", referenced from:
                _RAROpenArchiveEx in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                DataSet::DataSet() in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                _RARCloseArchive in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                DataSet::~DataSet() in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                SecPassword::~SecPassword() in libunrar_sys-a73d56025a07d327.rlib(secpassword.o)
                CmdExtract::CmdExtract(CommandData*) in libunrar_sys-a73d56025a07d327.rlib(extract.o)
                CmdExtract::~CmdExtract() in libunrar_sys-a73d56025a07d327.rlib(extract.o)
                ...
            "operator new(unsigned long)", referenced from:
                _RAROpenArchiveEx in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                SecPassword::SecPassword() in libunrar_sys-a73d56025a07d327.rlib(secpassword.o)
                CmdExtract::CmdExtract(CommandData*) in libunrar_sys-a73d56025a07d327.rlib(extract.o)
                RarCheckPassword::Set(unsigned char*, unsigned char*, unsigned int, unsigned char*) in libunrar_sys-a73d56025a07d327.rlib(extract.o)
                SecPassword::SecPassword(SecPassword const&) in libunrar_sys-a73d56025a07d327.rlib(extract.o)
                Archive::Archive(CommandData*) in libunrar_sys-a73d56025a07d327.rlib(archive.o)
                Archive::ReadHeader50() in libunrar_sys-a73d56025a07d327.rlib(arcread.o)
                ...
            "___cxa_begin_catch", referenced from:
                _RAROpenArchiveEx in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                _RARCloseArchive in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                _RARReadHeaderEx in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                ProcessFile(void*, int, char*, char*, wchar_t*, wchar_t*) in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                ___clang_call_terminate in libunrar_sys-a73d56025a07d327.rlib(extract.o)
            "___cxa_end_catch", referenced from:
                _RAROpenArchiveEx in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                _RARCloseArchive in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                _RARReadHeaderEx in libunrar_sys-a73d56025a07d327.rlib(dll.o)
                ProcessFile(void*, int, char*, char*, wchar_t*, wchar_t*) in libunrar_sys-a73d56025a07d327.rlib(dll.o)
          ld: symbol(s) not found for architecture x86_64
          clang-16: error: linker command failed with exit code 1 (use -v to see invocation)


error: could not compile `ouch` (bin "ouch") due to previous error

```

### Notify maintainers

@figsoda @psibi @nadir-ishiguro

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-darwin"`
 - host os: `Darwin 23.1.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `""`
 - channels(myuser): `""`
 - nixpkgs: `/nix/store/j2riph1av209fsm6crlgmbs3a38d875w-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/compression/ouch/default.nix
+++ b/pkgs/tools/compression/ouch/default.nix
@@ -7,6 +7,7 @@
 , xz
 , zlib
 , zstd
+, stdenv
 }:
 
 rustPlatform.buildRustPackage rec {
@@ -28,12 +29,19 @@ rustPlatform.buildRustPackage rec {
 
   buildFeatures = [ "zstd/pkg-config" ];
 
+  preCheck = ''
+    substituteInPlace tests/ui.rs \
+      --replace 'format!(r"/private{path}")' 'path.to_string()'
+  '';
+
   postInstall = ''
     installManPage artifacts/*.1
     installShellCompletion artifacts/ouch.{bash,fish} --zsh artifacts/_ouch
   '';
 
-  env.OUCH_ARTIFACTS_FOLDER = "artifacts";
+  env = { OUCH_ARTIFACTS_FOLDER = "artifacts"; } //
+    # Work around https://github.com/NixOS/nixpkgs/issues/166205.
+    lib.optionalAttrs stdenv.cc.isClang { NIX_LDFLAGS = "-l${stdenv.cc.libcxx.cxxabi.libName}"; };
 
   meta = with lib; {
     description = "A command-line utility for easily compressing and decompressing files and directories";

